### PR TITLE
Multiple domain support

### LINKED ISSUE #2790: Update middleware to support multiple domains
We want to allow Wagtail to determine which domain a request is coming in for so that it can serve the appropriate page. I've set up a new site on staging with a url of https://mozillafestival.mofostaging.net yet foundation.mofostaging.net is being served up. 

In Slack, @cadecairos mentioned this needs to be updated: https://github.com/mozilla/foundation.mozilla.org/blob/master/network-api/networkapi/utility/middleware.py#L11

### PR DESCRIPTION
**Pull request by cadecairos**
Closes #2790 

## Checklist

**Tests**
- [x] Is the code I'm adding covered by tests? - Maybe, can add tests but that might be a bit tough. I can look into this tomorrow morning

**Changes in Models:**
- [X] No changes to models

**Documentation:**
- [X] Is my code documented?
- [X] Did I update the READMEs or wagtail documentation?

### CODE REVIEW

File: source/js/main.js [hunk 1]
```diff
@@ -55,8 +55,12 @@ let main = {
     Array.from( document.querySelectorAll(`a`) ).forEach((link) => {
       let href = link.getAttribute(`href`);
 
+      function checkDomain(domain) {
+        return !href.match(`//${domain}`);
+      }
+
       // Define an external link as any URL with `//` in it
-      if (href && href.match(/\/\//) && !href.match(`//${env.TARGET_DOMAIN}`)) {
+      if (href && href.match(/\/\//) && env.TARGET_DOMAINS.some(checkDomain)) {
```
Comment (Pomax) [inline source/js/main.js]: if `checkDomain` is just a one-liner, we might as well use this:
```
if (href && href.match(`//`) && env.TARGET_DOMAINS.some(href => !href.match(`//${domain}`)) {
  ...
```
and dispense with the `checkDomain` function on L58.

File: docs/deployment.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ Production URL is [foundation.mozilla.org](https://foundation.mozilla.org)
 
 ### Domain Redirect
 
-Enable domain redirection by setting `DOMAIN_REDIRECT_MIDDLWARE_ENABLED` to `True`. This will enable a middleware function that checks every request, and return a 307 redirect to `TARGET_DOMAIN` if the host header does not match it.
+Enable domain redirection by setting `DOMAIN_REDIRECT_MIDDLWARE_ENABLED` to `True`. This will enable a middleware function that checks every request, and return a 307 redirect to the first listed domain in `TARGET_DOMAINS` if the host header does not match one of the domains specified there.
```
Comment (Pomax) [inline docs/deployment.md]: typo we never caught: `DOMAIN_REDIRECT_MIDDLWARE_ENABLED` should be `DOMAIN_REDIRECT_MIDDLEWARE_ENABLED`

File: env.default [hunk 3]
```diff
@@ -4,7 +4,7 @@ PULSE_API_DOMAIN=https://network-pulse-api-production.herokuapp.com
 PULSE_DOMAIN=www.mozillapulse.org
 # Always add a protocol (ex: https://) in front of the NETWORK_SITE_URL, including for localhost
 NETWORK_SITE_URL=https://foundation.mozilla.org
-TARGET_DOMAIN=foundation.mozilla.org
+TARGET_DOMAINS=foundation.mozilla.org
```
Comment (Pomax) [inline env.default]: this also needs a `DOMAIN_REDIRECT_MIDDLEWARE_ENABLED=False`, or django will crash, citing
```
Traceback (most recent call last):
  File ".virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\environ\environ.py", line 273, in get_value
    value = self.ENVIRON[var]
  File ".virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\os.py", line 678, in __getitem__
    raise KeyError(key) from None
KeyError: 'DOMAIN_REDIRECT_MIDDLEWARE_ENABLED'
```

**Comment by Pomax** [comment]
Redirects definitely work, just a few PR notes.

File: .travis.yml [hunk 4]
```diff
@@ -19,7 +19,7 @@ env:
   - RANDOM_SEED=530910203
   - SET_HSTS=False
   - SSL_REDIRECT=False
-  - TARGET_DOMAIN=foundation.mozilla.org
+  - TARGET_DOMAINS=foundation.mozilla.org
```
Comment (Pomax) [inline .travis.yml]: + DOMAIN_REDIRECT_MIDDLEWARE_ENABLED

File: appveyor.yml [hunk 5]
```diff
@@ -27,7 +27,7 @@ environment:
   PULSE_API_DOMAIN: https://network-pulse-api-production.herokuapp.com
   PULSE_DOMAIN: www.mozillapulse.org
   NETWORK_SITE_URL: https://foundation.mozilla.org
-  TARGET_DOMAIN: foundation.mozilla.org
+  TARGET_DOMAINS: foundation.mozilla.org
```
Comment (Pomax) [inline appveyor.yml]: + DOMAIN_REDIRECT_MIDDLEWARE_ENABLED

File: network-api/networkapi/settings.py [hunk 6]
```diff
@@ -92,9 +92,9 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = FILEBROWSER_DEBUG = env('DEBUG')
 
-# Force permanent redirects to the domain specified in TARGET_DOMAIN
-DOMAIN_REDIRECT_MIDDLWARE_ENABLED = env('DOMAIN_REDIRECT_MIDDLWARE_ENABLED')
-TARGET_DOMAIN = env('TARGET_DOMAIN')
+# Force permanent redirects to the domains specified in TARGET_DOMAINS
+DOMAIN_REDIRECT_MIDDLEWARE_ENABLED = env('DOMAIN_REDIRECT_MIDDLEWARE_ENABLED')
```
Comment (Pomax) [inline network-api/networkapi/settings.py]: is this missing a `DOMAIN_REDIRECT_MIDDLEWARE_ENABLED=(bool, False)` in the env block above? (would that obviate the need to specify it in env/confs?)

Comment (cadecairos) [reply network-api/networkapi/settings.py]: yeah, I think this is what we need

Comment (cadecairos) [reply network-api/networkapi/settings.py]: I'll add default env settings anyways.

**Comment by cadecairos** [comment]
@Pomax finally resolved the test issues. CI should go green.

**Pull request merged**